.github/workflows/build_and_test.yaml

```diff
@@ -259,7 +259,7 @@ jobs:
     - build
     - changes
     steps:
-    - uses: actions/checkout@v5.0.0
+    - uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8  # v5.0.0
```

Comment:
Isn't pinning to a tag much more readable? Are the tagged versions immutable?

Reply:
This specific format allows Dependabot to parse it info and accordingly raise a PR to bump the version to a newer one